dotnet/src/SemanticKernel.Abstractions/SkillDefinition/FunctionsView.cs

```diff
@@ -17,19 +18,21 @@ namespace Microsoft.SemanticKernel.SkillDefinition;
 [DebuggerDisplay("{DebuggerDisplay,nq}")]
 public sealed class FunctionsView
 {
+    private object _lock = new();
     /// <summary>
     /// Collection of semantic skill names and function names, including function parameters.
     /// Functions are grouped by skill name.
     /// </summary>
-    public ConcurrentDictionary<string, List<FunctionView>> SemanticFunctions { get; set; }
-        = new(StringComparer.OrdinalIgnoreCase);
+    public IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>> SemanticFunctions => this.GetFunctions(nativeFunctionCheck: false);
 
     /// <summary>
     /// Collection of native skill names and function views, including function parameters.
     /// Functions are grouped by skill name.
     /// </summary>
-    public ConcurrentDictionary<string, List<FunctionView>> NativeFunctions { get; set; }
-        = new(StringComparer.OrdinalIgnoreCase);
+    public IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>> NativeFunctions => this.GetFunctions(nativeFunctionCheck: true);
+
+    private readonly ConcurrentDictionary<string, List<FunctionView>> _semanticFunctions = new(StringComparer.OrdinalIgnoreCase);
```

Comment:
We have no requirement for these to be Concurrent, and the rest of the Kernel is not built for that either. Can we just drop the overhead of concurrency, and just make it a Dictionary? This is just a Functions*View* - a snapshot of the functions - and shouldn't be modified by multiple (or even one) thread after creation anyway.

Reply:
> Can we just drop the overhead of concurrency, and just make it a Dictionary?

+1. We should move in that direction everywhere we don't absolutely need the concurrency guarantees (especially since they often require more than just a different dictionary type).